### PR TITLE
[PTen]Elementwise_div Kernel Refactor

### PR DESCRIPTION
**Pull request by YuanRisheng**
### PR types

Others
### PR changes

OPs
### Describe

move elementwise_div kernel from fluid_dir to pten_dir and refactor it.

### CODE REVIEW

File: paddle/pten/tests/kernels/test_elementwise_dev_api.cc [hunk 1]
```diff
@@ -129,3 +129,56 @@ TEST(DEV_API, subtract) {
...
+      *(static_cast<paddle::platform::CPUDeviceContext*>(dev_ctx)),
+      dense_x,
+      dense_y,
+      axis);
+
+  // 3. check result
+  ASSERT_EQ(dense_out.dims().size(), 2);
+  ASSERT_EQ(dense_out.dims()[0], 3);
+  ASSERT_EQ(dense_out.meta().dtype, pten::DataType::FLOAT32);
+  ASSERT_EQ(dense_out.meta().layout, pten::DataLayout::NCHW);
```
Comment (zyfncg) [inline paddle/pten/tests/kernels/test_elementwise_dev_api.cc:175]: Maybe `dense_out.dtype()` and `dense_out.layout()` is better

**Review by zyfncg — APPROVED** [review]

File: paddle/pten/kernels/cpu/math.cc [hunk 2]
```diff
@@ -114,6 +114,30 @@ void ElementwiseSub(const CPUContext& dev_ctx,
   }
 }
 
+template <typename T>
+void ElementwiseDiv(const CPUContext& dev_ctx,
```
Comment (chenwhql) [inline paddle/pten/kernels/cpu/math.cc:118]: 中间层kernel API和底层kernel API的命名是不是一致比较好，可以后续统一更改下

**Review by chenwhql — APPROVED** [review]
LGTM

File: paddle/pten/kernels/cpu/math.cc (hunk 2 above)
Comment (YuanRisheng) [reply paddle/pten/kernels/cpu/math.cc:118]: 好的，下个pr改一下

File: paddle/pten/tests/kernels/test_elementwise_dev_api.cc (hunk 1 above)
Comment (YuanRisheng) [reply paddle/pten/tests/kernels/test_elementwise_dev_api.cc:175]: OK

**Review by Shixiaowei02 — APPROVED** [review]

**Review by MingMingShangTian — APPROVED** [review]
LGTM

**Pull request merged**